internal/pkg/cli/project_delete.go

```diff
@@ -0,0 +1,251 @@
+
+var (
+	errOperationCancelled = errors.New("operation cancelled")
+)
+
+// BuildProjectDeleteCommand builds the `project delete` subcommand.
+func BuildProjectDeleteCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete all resources associated with the local project.",
+		Example: `
+  /code $ ecs-preview project delete --yes`,
+		RunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			opts, err := newDeleteProjOpts()
+			if err != nil {
+				return err
+			}
+
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+
+			if err := opts.Ask(); err != nil {
+				return err
+			}
+
+			return opts.Execute()
+		}),
+	}
+
+	cmd.Flags().Bool(yesFlag, false, yesFlagDescription)
+	viper.BindPFlag(yesFlag, cmd.Flags().Lookup(yesFlag))
+	cmd.Flags().String(profileFlag, defaultProfile, profileFlagDescription)
+	viper.BindPFlag(profileFlag, cmd.Flags().Lookup(profileFlag))
+
+	return cmd
+}
+
+type deleteProjOpts struct {
+	*GlobalOpts
+
+	projectName      string
+	skipConfirmation bool
+	profile          string
+
+	sessProvider sessionProvider
+	store        projectService
+	deployer     deployer
+	ws           workspaceDeleter
+	spinner      progress
+}
+
+type workspaceDeleter interface {
+	Delete() error
+}
+
+func newDeleteProjOpts() (*deleteProjOpts, error) {
+	store, err := store.New()
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+
+	s, err := session.NewProvider().Default()
+	if err != nil {
+		return nil, err
+	}
+	cf := cloudformation.New(s)
+
+	return &deleteProjOpts{
+		GlobalOpts:       NewGlobalOpts(),
+		projectName:      viper.GetString(projectFlag),
+		skipConfirmation: viper.GetBool(yesFlag),
+		profile:          viper.GetString(profileFlag),
+		store:            store,
+		ws:               ws,
+		deployer:         cf,
+		spinner:          termprogress.NewSpinner(),
+	}, nil
+}
+
+func (o deleteProjOpts) Validate() error {
+	if o.ProjectName() == "" {
+		return errNoProjectInWorkspace
+	}
+
+	return nil
+}
+
+func (o deleteProjOpts) Ask() error {
+	if o.skipConfirmation {
+		return nil
+	}
+
+	manualConfirm, err := o.prompt.Confirm(
+		fmt.Sprintf(fmtConfirmProjectDeletePrompt, o.ProjectName()),
+		confirmProjectDeleteHelp,
+		prompt.WithTrueDefault())
+
+	if err != nil {
+		return err
+	}
+
+	if !manualConfirm {
+		return errOperationCancelled
+	}
+
+	return nil
+}
+
+func (o deleteProjOpts) Execute() error {
+	if err := o.deleteApps(); err != nil {
+		return err
+	}
+
+	envs, err := o.store.ListEnvironments(o.ProjectName())
+	if err != nil {
+		return err
+	}
+
+	accounts := newStringSet()
+	regions := newStringSet()
+
+	for _, e := range envs {
+		accounts.put(e.AccountID)
+		regions.put(e.Region)
+	}
+
+	// NOTE: Ordering of project infrastructure stackset and environment deletion is significant.
+	// If we delete environments first we lose account and region information necessary to delete
+	// the stacksets.
```

Comment:
If we already queried for `store.ListEnvironments` before deleting the environments or project stackset how do lose that information? Am I missing something?

Reply:
If the process is cancelled out for any reason before initiating stackset instance deletion then we're kind of stuck since we've lost the AccountID and Region information.

Reply:
Makes sense! can we capture your reply in the comments?